libindy/src/commands/non_secrets.rs

```diff
@@ -303,6 +366,256 @@ impl NonSecretsCommandExecutor {
+                    })
+                )
+            )
+        ).unwrap();
+    }
+
+    fn _get_cred_def_continue(&self, wallet_handle: WalletHandle, ledger_response: IndyResult<(String, String)>, options: GetCacheOptions, cb_id: i32) {
+        let cb = self.pending_callbacks.borrow_mut().remove(&cb_id).expect("FIXME INVALID STATE");
+
+        let (cred_def_id, cred_def_json) = try_cb!(ledger_response, cb);
+
+        if !options.no_store.unwrap_or(false) {
+            let mut tags = Tags::new();
+            let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i32;
+            tags.insert("timestamp".to_string(), ts.to_string());
+            let _ = self.wallet_service.delete_record(wallet_handle, CRED_DEF_CACHE, &cred_def_id);
+            let _ = self.wallet_service.add_record(wallet_handle, CRED_DEF_CACHE, &cred_def_id, &cred_def_json, &tags);
+        }
+
+        cb(Ok(cred_def_json));
+    }
+
+    fn purge_schema_cache(&self,
+                          wallet_handle: WalletHandle,
+                          options_json: &str) -> IndyResult<()> {
+        trace!("purge_schema_cache >>> wallet_handle: {:?}, options_json: {:?}", wallet_handle, options_json);
+
+        let options = serde_json::from_str::<PurgeOptions>(options_json)
+            .to_indy(IndyErrorKind::InvalidStructure, "Cannot deserialize options")?;
+
+        let max_age = options.max_age.unwrap_or(-1);
+        let query_json = if max_age >= 0 {
+            let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i32;
+            json!({"timestamp": {"$lt": ts - max_age}}).to_string()
+        } else {
+            "{}".to_string()
+        };
+
+        let options_json = json!({
+            "retrieveType": false,
+            "retrieveValue": false,
+            "retrieveTags": false,
+        }).to_string();
+
+        let mut search = self.wallet_service.search_records(
+            wallet_handle,
+            SCHEMA_CACHE,
+            &query_json,
+            &options_json,
+        )?;
+
+        while let Some(record) = search.fetch_next_record()? {
+            self.wallet_service.delete_record(wallet_handle, SCHEMA_CACHE, record.get_id())?;
+        }
+
+        let res = ();
+
+        trace!("purge_schema_cache <<< res: {:?}", res);
+
+        Ok(res)
+    }
+
+    fn purge_cred_def_cache(&self,
+                            wallet_handle: WalletHandle,
+                            options_json: &str) -> IndyResult<()> {
+        trace!("purge_cred_def_cache >>> wallet_handle: {:?}, options_json: {:?}", wallet_handle, options_json);
+
+        let options = serde_json::from_str::<PurgeOptions>(options_json)
+            .to_indy(IndyErrorKind::InvalidStructure, "Cannot deserialize options")?;
+
+        let max_age = options.max_age.unwrap_or(-1);
+        let query_json = if max_age >= 0 {
+            let ts = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs() as i32;
```

Comment:
unwrap here may fail in theory on windows, isn't it? In this case I suggest to handle the error (may be as invalid state or even ignore somehow, not sure)

Reply:
In documentation is says: Returns an [`Err`] if `earlier` is later than `self`, and the error contains how far from `self` the time is.
I may return error, if I ignore it would work different than user expects.